src/SignalR/clients/csharp/Http.Connections.Client/src/HttpConnection.cs

```diff
@@ -148,6 +148,8 @@ public HttpConnection(HttpConnectionOptions httpConnectionOptions, ILoggerFactor
 
         _url = _httpConnectionOptions.Url;
 
+        RemoteEndPoint = new UriEndPoint(_url);
```

Comment:
I wonder if this should be set after negotiate finishes, since negotiate can redirect to another endpoint, e.g. Azure SignalR Service

Reply:
Right now this actually isn’t used. The activity uses values from _endpoint

Reply:
How does negotiate change the url? Is there an example in tests of this happening?

The invocation activity is started before the connection is acquired. I could change logic to update server.address and server.port if the connection reports different value.

Reply:
https://github.com/dotnet/aspnetcore/blob/26ee437e46d8cdbea7113833942690c2ad2a28c8/src/SignalR/clients/csharp/Client/test/UnitTests/HttpConnectionTests.Negotiate.cs#L415

Spec that talks about it:
https://github.com/dotnet/aspnetcore/blob/main/src/SignalR/docs/specs/TransportProtocols.md#all-versions

Reply:
Done. Client span's now report the negotiated URL.

[`a698667` (#57101)](https://github.com/dotnet/aspnetcore/pull/57101/commits/a698667a9478e6be87395eb4bc9987b4158e7c15)